core/src/test/java/kafka/admin/DescribeAuthorizedOperationsTest.java

```diff
@@ -0,0 +1,275 @@
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static org.apache.kafka.common.acl.AccessControlEntryFilter.ANY;
+import static org.apache.kafka.common.acl.AclOperation.ALL;
+import static org.apache.kafka.common.acl.AclOperation.ALTER;
+import static org.apache.kafka.common.acl.AclOperation.DELETE;
+import static org.apache.kafka.common.acl.AclOperation.DESCRIBE;
+import static org.apache.kafka.common.acl.AclPermissionType.ALLOW;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+@ExtendWith(ClusterTestExtensions.class)
+public class DescribeAuthorizedOperationsTest {
+    private static final String GROUP1 = "group1";
+    private static final String GROUP2 = "group2";
+    private static final String GROUP3 = "group3";
+    private static final ResourcePattern GROUP1_PATTERN = new ResourcePattern(ResourceType.GROUP, GROUP1, PatternType.LITERAL);
+    private static final ResourcePattern GROUP2_PATTERN = new ResourcePattern(ResourceType.GROUP, GROUP2, PatternType.LITERAL);
+    private static final ResourcePattern GROUP3_PATTERN = new ResourcePattern(ResourceType.GROUP, GROUP3, PatternType.LITERAL);
+    private static final ResourcePattern CLUSTER_PATTERN = new ResourcePattern(ResourceType.CLUSTER, Resource.CLUSTER_NAME, PatternType.LITERAL);
+    private static final AccessControlEntry ALTER_ENTRY = createAccessControlEntry(JaasUtils.KAFKA_PLAIN_USER1, ALTER);
+    private static final AccessControlEntry DESCRIBE_ENTRY = createAccessControlEntry(JaasUtils.KAFKA_PLAIN_USER1, DESCRIBE);
+
+    static List<ClusterConfig> generator() {
+        return List.of(
+            ClusterConfig.defaultBuilder()
+                .setTypes(Set.of(Type.KRAFT))
+                .setServerProperties(Map.of(GroupCoordinatorConfig.OFFSETS_TOPIC_PARTITIONS_CONFIG, "1"))
+                .setServerProperties(Map.of(GroupCoordinatorConfig.OFFSETS_TOPIC_REPLICATION_FACTOR_CONFIG, "1"))
+                .setBrokerSecurityProtocol(SecurityProtocol.SASL_PLAINTEXT)
+                .setControllerSecurityProtocol(SecurityProtocol.SASL_PLAINTEXT)
+                .build()
+        );
+    }
+
+    private static AccessControlEntry createAccessControlEntry(String username, AclOperation operation) {
+        return new AccessControlEntry(
+            new KafkaPrincipal(KafkaPrincipal.USER_TYPE, username).toString(),
+            AclEntry.WILDCARD_HOST,
+            operation,
+            ALLOW
+        );
+    }
+
+    private Map<String, Object> createAdminConfig(String username, String password) {
+        return createAdminConfig(username, password, Map.of());
+    }
+
+    private Map<String, Object> createAdminConfig(String username, String password, Map<String, Object> additionalConfigs) {
```

Comment:
Not sure `createAdminConfig()` is the best name since this is also used to create consumer configs. What about `createClientConfig()` or `createConfig()`?

Reply:
We use `alterConsumerGroupOffsets` to create group, so we don't need this function. Thanks!